viewer/packages/tools/src/Geomap/GeomapTool.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import { Cognite3DViewer } from '@reveal/core';
+import { Cognite3DViewerToolBase } from '../Cognite3DViewerToolBase';
+import { MapConfig, MapProviders, MapboxStyle } from './MapConfig';
+import { Geomap } from './Geomap';
+/**
+ * The `GeomapTool` is a geolocation for the models and allow the user to place them on the maps.
+ */
+export class GeomapTool extends Cognite3DViewerToolBase {
+  private readonly _viewer: Cognite3DViewer;
+  private readonly _maps: Geomap;
+
+  constructor(viewer: Cognite3DViewer, config: MapConfig) {
+    super();
+
+    this._viewer = viewer;
+    this._maps = new Geomap(this._viewer, config);
+  }
+
+  public setMapProvider(provider: MapProviders, apiKey: string, appCode?: string, id?: MapboxStyle) {
+    this._maps.setMapProvider(provider, apiKey, appCode, id);
+  }
+
+  public dispose(): void {
+    super.dispose();
+  }
```

Comment:
Do we need this function? I think the tool can have a static provider and don't need the ability to set this runtime - only on construction. This function seems a bit off and I don't like that it has a MapboxStyle-argument which only makes sense if provider is Mapbox.

Reply:
Creating providers objects static to use them would take more memory (3 provider objects) than creating provider at runtime (1 provider object). I assume the users will only use one map (provider), so it would be easier to maintain single provider.
As mentioned earlier `GeomapTool.setMapProvider` is useful for only example to showcase all 3 map providers. If we do not want the 3 option in examples, we can remove it, but it won't be useful for the user.